prow/cmd/branchprotector/protect.go

```diff
@@ -63,45 +62,22 @@ func gatherOptions() options {
 	return o
 }
 
-func jobRequirements(jobs []config.Presubmit, after bool) []string {
-	var required []string
-	for _, j := range jobs {
-		// Does this job require a context or have kids that might need one?
-		if !after && !j.AlwaysRun && j.RunIfChanged == "" {
-			continue // No
-		}
-		if !j.SkipReport { // This job needs a context
-			required = append(required, j.Context)
-		}
-		// Check which children require contexts
-		required = append(required, jobRequirements(j.RunAfterSuccess, true)...)
-	}
-	return required
-}
-
-func repoRequirements(org, repo string, cfg config.Config) []string {
```

Comment:
A second thing would be to write a `allBranchRequirements(org, repo, branch string, cfg config.Confg)` that combines `repoRequirements()` with the union of all the contexts at branch, repo, org, overall levels...

For this PR can we focus on making these minimal set of changes and save refactoring the whole thing for later?